src/include/gurt/common.h

```diff
@@ -204,12 +204,50 @@ extern "C" {
 #define D_ALLOC_ARRAY(ptr, count) D_ALLOC_CORE(ptr, sizeof(*ptr), count)
 #define D_FREE_PTR(ptr)		D_FREE(ptr)
 
-#define D_GOTO(label, rc)			\
-	do {					\
-		__typeof__(rc) __rc = (rc);		\
-		(void)(__rc);			\
-		goto label;			\
+/* _Generic() is part of the c11 standard but isn't yet supported by
+ * gcc until version 4.9 so check for that but enable it in all other
+ * cases.
+ */
+
+#ifdef __GNU__
+#if __GNUC__ > 4 || \
+	(__GNUC__ == 4 && (__GNUC_MINOR__ > 9 ||))
+#define VERBOSE_GOTO 0
+#else
+#define VERBOSE_GOTO 1
+#endif
+#define VERBOSE_GOTO 1
```

Comment:
should there be another else here?   It seems we always set it to 1 if __GNU__ is set

Reply:
I'm assuing here all compilers other than GCC 4.8 and below do support it, there are two #endifs here.